desktop/core/src/desktop/lib/botserver/views_tests.py

```diff
@@ -75,34 +75,66 @@ def test_handle_on_message(self):
   def test_handle_on_link_shared(self):
     with patch('desktop.lib.botserver.views.slack_client.chat_unfurl') as chat_unfurl:
       with patch('desktop.lib.botserver.views._make_unfurl_payload') as mock_unfurl_payload:
-        with patch('desktop.lib.botserver.views.Document2.objects.get') as document2_objects_get:
-          with patch('desktop.lib.botserver.views._get_gist_document') as _get_gist_document:
-            with patch('desktop.lib.botserver.views.send_result_file') as send_result_file:
+        with patch('desktop.lib.botserver.views._get_gist_document') as _get_gist_document:
+          with patch('desktop.lib.botserver.views.send_result_file') as send_result_file:
+            with patch('desktop.lib.botserver.views.slack_email_prefix') as email_prefix:
 
+              # Slack user email: test@example.com
               client = make_logged_in_client(username="test", groupname="default", recreate=True, is_superuser=False)
               user = User.objects.get(username="test")
+              # Other slack user email: test_not_me@example.com
+              client_not_me = make_logged_in_client(username="test_not_me", groupname="default", recreate=True, is_superuser=False)
+              user_not_me = User.objects.get(username="test_not_me")
+
               channel_id = "channel"
               message_ts = "12.1"
+              user_id = "<@user_id>"
 
-              # qhistory link
-              links = [{"url": "https://demo.gethue.com/hue/editor?editor=123456"}]
+              # Query link
+              links = [{"url": "https://demo.gethue.com/hue/editor?editor=12345"}]
               doc_data = {
                 "dialect": "mysql",
                 "snippets": [{
                   "database": "hue",
                   "statement_raw": "SELECT 5000",
                 }]
               }
-              document2_objects_get.return_value = Mock(data=json.dumps(doc_data), owner=user)
+              doc = Document2.objects.create(id=12345, data=json.dumps(doc_data), owner=user)
               mock_unfurl_payload.return_value = {
                 'payload': {},
                 'file_status': True,
               }
-              handle_on_link_shared(channel_id, message_ts, links)
+
+              # Other user sends link to unfurl without read permission
+              email_prefix.return_value = 'test_not_me'
```

Comment:
This should not be mocked, but tested (whole point of the tests :)

Mock the return of `slack_client.users_info` instead

Reply:
Fixed